File: backend/routes/authRoutes.py
from __future__ import annotations

from flask import Blueprint, jsonify, request

from backend.db import execute_query
from backend.utils.jwt_helper import create_access_token
from backend.utils.password import hash_password_sha256


auth_bp = Blueprint("auth", __name__)


@auth_bp.post("/login")
def login():
    """POST: user login (student, admin)"""
    payload = request.get_json(silent=True) or {}
    email = (payload.get("email") or "").strip().lower()
    password = payload.get("password")

    if not email or not password:
        return jsonify({"success": False, "message": "Email and password are required"}), 400

    # Hash password using SHA256 (matches crypto-js.SHA256)
    hash_password = hash_password_sha256(password)

    # Query user with email and hashed password
    sql = "SELECT * FROM users WHERE email = %s AND password = %s"
    results = execute_query(sql, (email, hash_password))

    # If user does not exist
    if not results or len(results) == 0:
        return jsonify({"success": False, "message": "Invalid email or password!"}), 401

    # If user exists
    user = results[0]

    # Create token
    token = create_access_token(user["email"], {"role": user.get("role", "student")})

    return jsonify({
        "success": True,
        "message": "login successful",
        "data": {"token": token}
    }), 200

File: express-backend/routes/authRoutes.js
const router = require("express").Router();
const { USER_TABLE, JWT_SECRET_KEY } = require("../config");
const { pool } = require("../db-connection/db");
const { successResponse, errorResponse } = require("../utils/apiResponse");
const cryptoJs = require("crypto-js");
const jwt = require("jsonwebtoken");

// POST: user login (student, admin)
router.post("/login", (request, response) => {
  const { email, password } = request.body;

  const hashPassword = String(cryptoJs.SHA256(password));

  const sql = `SELECT * FROM ${USER_TABLE} WHERE email = ? AND password = ?`;

  pool.query(sql, [email, hashPassword], (error, results) => {
    if (error) return response.send(errorResponse(error));

    // if user does not exists
    if (results.length === 0)
      return response.send(errorResponse("Invalid email or password!"));

    // if user exists
    const user = results[0];

    // creating token
    const token = jwt.sign(
      { email: user.email, role: user.role },
      JWT_SECRET_KEY
    );

    const message = {
      message: "login successful",
      token,
    };

    return response.send(successResponse(message));
  });
});

module.exports = router;
